tests/scripts/run_demos.py

```diff
@@ -1,44 +1,57 @@
 #!/usr/bin/env python3
 """Run the Mbed TLS demo scripts.
 """
+import argparse
 import glob
 import subprocess
 import sys
 
-def run_demo(demo):
+def run_demo(demo, quiet=False):
     """Run the specified demo script. Return True if it succeeds."""
-    returncode = subprocess.call([demo])
+    args = {}
+    if quiet:
+        args['stdout'] = subprocess.DEVNULL
+        args['stderr'] = subprocess.DEVNULL
+    returncode = subprocess.call([demo], **args)
     return returncode == 0
 
-def run_demos(demos):
+def run_demos(demos, quiet=False):
     """Run the specified demos and print summary information about failures.
 
     Return True if all demos passed and False if a demo fails.
     """
     failures = []
     for demo in demos:
-        print('#### {} ####'.format(demo))
-        success = run_demo(demo)
+        if not quiet:
```

Comment:
Thanks for adding a `-q` option, but it's more quiet than I was wishing, I should have been more explicit: I'd like the option only suppress stdout from the demos themselves, not the output from this script, which is actually what I want to see (not lost in the middle of the demos' stdout).

(What I had in mind is what many of our scripts do: `run_tests.pl` as well as `ctest` show you the list of test suites run but not their output, `ssl-opt.sh` and `compat.sh` don't bother your with the output of `ssl_client2`, etc.)

Reply:
This is turning a bit into scope creep, but ok, done.

Reply:
Ok, I'm sorry, apparently I'm expressing myself very poorly here. I'd be 100% if `-q` had just suppressed stdout (and as you point out, stderr too for consistency) from the demos and changed absolutely nothing else elsewhere. (For reference, my original request was " a -q option where stdout of the demos is suppressed.")

My reference to other scripts was only for the "show you what they run, but not the output of what they run" aspect, not a request to copy the format. I just wanted to point out that wanting to see just what's being run seemed common. I should have made that clear.

So at this point I'm hesitating between:

- requesting to change to `-q` to what I was originally asking, in order to avoid cluttering the code with the result of our misunderstandings
- not requesting any change and approving the code as it because I'm afraid these misunderstanding have already wasted more of your time than this option deserves.

So I'll leave it up to you.

Reply:
I think with each message I understand less what you want `-q` to do.

How about we stop here and you make a follow-up that tunes `-q` to your preference?

Reply:
Sure. It's probably not worth a follow-up PR, as `run_demos.py` is simple enough that the code clutter is probably not an issue, but for reference, here's what I had in mind: https://github.com/mpg/mbedtls/blob/2698-q/tests/scripts/run_demos.py